h/static/scripts/annotation-mapper.js

```diff
@@ -18,11 +18,29 @@ function getContainer(threading, annotation) {
 
 // Wraps the annotation store to trigger events for the CRUD actions
 // @ngInject
-function annotationMapper($rootScope, threading, store) {
+function annotationMapper($rootScope, threading, store, groups) {
+
+  // All of the annotations that annotationMapper has received from calls to
+  // its loadAnnotations() method.
+  var received = [];
+
+  // The annotations that are currently loaded into the page context.
+  var loaded = [];
```

Comment:
We could maybe avoid having to keep track of this list here, by just getting it from `threading`. I sort of feel like it's `annotationMapper`'s job to keep track of this though, and `threading` is just a dump.